src/util/error-extractor/index.js

```diff
@@ -0,0 +1,68 @@
+
+class ErrorDetailsExtractor {
+  constructor(extractObjBuilder) {
+    this.status=extractObjBuilder.status;
+    this.messageDetails = { ...extractObjBuilder.messageDetails }
+  }
+
+  static get Builder() {
+    return ErrorDetailsExtractBuilder;
+  }
+  
+}
+
+
+class ErrorDetailsExtractBuilder {
+
+  constructor() {
+    this.status = 0;
```

Comment:
Why can't just have 3 fields - `status`, `field`, `message` ?

Reply:
If `messageDetails` is intended to be an array, then `field`:`message` allows for extension of multiple key-value pairs

Reply:
The parameter field is related to error message.
We will extract whatever message is available in the field
For example
If the response from facebook_pixel looks like this
```
error": {
    "message": "Invalid parameter",
    "type": "OAuthException",
    "code": 100,
    "error_subcode": 2804007,
    "is_transient": false,
    "error_user_title": "Invalid IP Address",
    "error_user_msg": "The IP address you entered is invalid. Enter a valid IP address.",
    "fbtrace_id": "AZN7qSAksRz48Xdc_YNQ9_7"
}

Here we can see clearly that if we use the content in `error_user_title` field, that should sufficiently provide us context about the error message itself
```
Since field is related to message, I thought it is best to put in `messageDetails` object, making it easy to understand.

>Why can't just have 3 fields - status, field, message ?

`field` is only related to error message(in this case `message`). Hence having them would not make it obvious about the usage of `field`